dbt/models/default/default.vw_pin_sale.sql

```diff
@@ -249,93 +249,204 @@ sales_val AS (
         sf.run_id AS sv_run_id,
         sf.version AS sv_version
     FROM
-        {{ source('sale', 'flag') }}
-            AS sf
+        {{ source('sale', 'flag') }} AS sf
     INNER JOIN max_version_flag AS mv
         ON sf.meta_sale_document_num = mv.meta_sale_document_num
         AND sf.version = mv.max_version
+),
+
+-- CTE to coalesce iasworld and mydec values prior to
+-- constructing filters that depend on coalesced fields
+combined_sales AS (
+    SELECT
+        COALESCE(uq_sales.pin, md_sales.pin) AS pin_coalesced,
+        -- For many of the fields we used  simple coalesce statement,
+        -- but some data is a bit more complicated. Prior to 2021,
+        -- mydec sales and iasworld sales used different sale dates.
+        -- We preference the mydec sale as they are believed to be more
+        -- accurate. As of 2021, iasworld utilizes mydec sales, which means
+        -- we can prioritize iasworld data instead of mydec data.
+        CASE
```

Comment:
I think `year_coalesced` and `sale_date_coalesced` should follow the exact same logic - let's make sure `year_coalesced` respects the 2021 cut-off